packages/strapi-plugin-upload/admin/src/components/BrowseAssets/index.js

```diff
@@ -113,7 +113,7 @@ const BrowseAssets = () => {
             data={files}
             onChange={handleFileSelection}
             selectedItems={selectedFiles}
-            onCardClick={handleListCardClick}
+            onCardClick={canSelectFile && handleListCardClick}
```

Comment:
Here you are returning a boolean instead of a function if `canSelectFile` is false.
You should handle the statement in the `handleListCardClick`:

```
const handleListCardClick = id => {
 if (!canSelectFile) {
   return;
 }

 // ...
}
```